oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -110,6 +111,8 @@ public class ImpersonatedCredentials extends GoogleCredentials
 
   private transient HttpTransportFactory transportFactory;
 
+  @VisibleForTesting transient Calendar calendar = Calendar.getInstance();
```

Comment:
This should be private and implemented via builder. 

Maybe you can use the same approach as custom lifetime in ServiceAccountCredentials: https://github.com/googleapis/google-auth-library-java/blob/c21ec6c952b8bb8fb8bc2e2f1b260beb330a3cd2/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L666

Reply:
Updated